src/schema/schema_drop.c

```diff
@@ -234,8 +237,13 @@ __drop_tiered(WT_SESSION_IMPL *session, const char *uri, bool force, const char
         WT_ERR(__wt_tiered_name(session, &tiered->iface, i, WT_TIERED_NAME_OBJECT, &name));
         __wt_verbose(session, WT_VERB_TIERED, "DROP_TIERED: remove object %s from metadata", name);
         WT_ERR_NOTFOUND_OK(__wt_metadata_remove(session, name), false);
-        if (remove_files && tier != NULL)
-            WT_TRET(__wt_meta_track_drop(session, tier->name));
+        if (remove_files && tier != NULL) {
+            filename = name;
+            WT_PREFIX_SKIP_REQUIRED(session, filename, "object:");
+            WT_ERR(__wt_fs_exist(session, filename, &exist));
+            if (exist)
+                WT_TRET(__wt_meta_track_drop(session, filename));
```

Comment:
Same here, why `WT_TRET` and not `WT_ERR`?  Won't any error that happens here be effectively ignored as `ret` will be overwritten further down?  Also, I see another call to `WT_TRET` in this function that isn't part of this review.

Reply:
Thanks. Originally this function was modeled on `drop_file` which uses WT_TRET. But with the WT_ERR we cannot get to that line with an error so WT_ERR is correct. Good catch. I'll fix those uses.